boot/freeldr/bootsect/ext.S

```diff
@@ -0,0 +1,490 @@
+    jne PrintDiskError
+    test cl, 1                              // si = API subset support bitmap
+    jz PrintDiskError                       // Bit 0, extended disk access functions (AH=42h-44h,47h,48h) supported
+
+LoadExtraBootCode:
+    // First we have to load our extra boot code at
+    // sector 1 into memory at [0000:7e00h]
+    xor eax, eax
+    inc eax
+    mov cx, 1
+    xor bx, bx
+    mov es, bx                                          // Read sector to [0000:7e00h]
+    mov bx, HEX(7e00)
+    call ReadSectors
+
+    jmp  LoadRootDirectory
+
+// Reads logical sectors into ES:[BX]
+// EAX has logical sector number to read
+// CX has number of sectors to read
+ReadSectors:
+    push es
+    add  eax, dword ptr BP_REL(ExtVolumeStartSector)	// Add the start of the volume
+                                                // If at all possible we want to use LBA routines because
+                                                // They are optimized to read more than 1 sector per read
+
+ReadSectorsLBA:
+    pushad                                      // Save logical sector number & sector count
+
+    cmp  cx, 64                                 // Since the LBA calls only support 0x7F sectors at a time we will limit ourselves to 64
+    jbe  ReadSectorsSetupDiskAddressPacket      // If we are reading less than 65 sectors then just do the read
+    mov  cx, 64                                 // Otherwise read only 64 sectors on this loop iteration
+
+ReadSectorsSetupDiskAddressPacket:
+    mov word ptr ds:[LBASectorsRead],cx
+    push 0
+    push 0
+    push eax                                // Put 64-bit logical block address on stack
+    push es                                 // Put transfer segment on stack
+    push bx                                 // Put transfer offset on stack
+    push cx                                 // Set transfer count
+    push 16                                 // Set size of packet to 10h
+    mov  si, sp                             // Setup disk address packet on stack
+
+    mov  dl, byte ptr BP_REL(BootDrive)     // Drive number
+    mov  ah, HEX(42)                        // Int 13h, AH = 42h - Extended Read
+    int  HEX(13)                            // Call BIOS
+    jc   PrintDiskError                     // If the read failed then abort
+
+    add  sp, 16                             // Remove disk address packet from stack
+
+    popad                                   // Restore sector count & logical sector number
+
+    push bx
+    mov  ebx, dword ptr ds:[LBASectorsRead]
+    add  eax, ebx                           // Increment sector to read
+    shl  ebx, 5
+    mov  dx, es
+    add  dx, bx                             // Setup read buffer for next sector
+    mov  es, dx
+    pop  bx
+
+    sub  cx, word ptr ds:[LBASectorsRead]
+    jnz  ReadSectorsLBA                     // Read next sector
+
+    pop es
+    ret
+
+LBASectorsRead:
+    .long 0
```

Comment:
Can you please move this variable definition to below the `ExtFileIndirectPointerDouble`?

Reply:
Done! i moved.